cgi/Project.pm

```diff
@@ -10,9 +10,11 @@ sub new
   $self->{$_} = $args{$_} for keys %args;
   my $id = $args{'id'};
   die "No CRMS object passed to project" unless $args{'crms'};
-  my $sql = 'SELECT * FROM projects WHERE id=?';
-  my $ref = $self->{'crms'}->get('dbh')->selectall_hashref($sql, 'id', undef, $id);
-  $self->{$_} = $ref->{$id}->{$_} for keys %{$ref->{$id}};
+  if (defined $id) {
```

Comment:
I added this so we could create a Project instance without an id (primary key). Normally we never instantiate the Project class, just the subclasses, but this gives me a chance to test the default behavior without writing tests for all of the subclasses (definitely out of scope for this ticket).